src/versionedRouter.js

```diff
@@ -487,11 +483,23 @@ async function routerHandleDest(ctx) {
             tags.FEATURES.ROUTER,
           );
         }
-        if (routerDestHandler.processMetadataForRouter) {
-          listOutput.forEach((output) => {
-            output.metadata = routerDestHandler.processMetadataForRouter(output);
-          });
-        }
+        const hasProcMetadataForRouter = routerDestHandler.processMetadataForRouter;
+        // enriching transformed event
+        listOutput.forEach(listOut => {
+          const { batchedRequest } = listOut;
+          if (Array.isArray(batchedRequest)) {
+            // eslint-disable-next-line no-param-reassign
+            listOut.batchedRequest = batchedRequest.map(batReq => enrichTransformedEvent(batReq));
+          } else if (batchedRequest && typeof batchedRequest === 'object') {
+            // eslint-disable-next-line no-param-reassign
+            listOut.batchedRequest = enrichTransformedEvent(batchedRequest);
+          }
+
+          if (hasProcMetadataForRouter) {
+            // eslint-disable-next-line no-param-reassign
+            listOut.metadata = routerDestHandler.processMetadataForRouter(listOut);
+          }
+        });
```

Comment:
This piece of code is not clearly understandable. Cannot clearly make-out what are we trying to do here?
can we reduce the complexity of this code in someway?

Reply:
This code is basically dealing with the case when batched request comes in form of array (ex: Marketo) or in form of object (ex: Mp). At last for every request we are processing metadata if it needs to be.

Reply:
ok